src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
```

Comment:
The `die` method uses `indexOf` and `splice` to remove the animal from the `Animal.alive` array, which is not optimal for performance, especially if there are many animals in the array. Consider using `filter` to avoid the performance cost of nesting loops.